### PR TITLE
Implement Create Port Game Command

### PR DESCRIPTION
**Pull request by duncanspumpkin**
Seems to work. Haven't done replay tests on it yet.

### CODE REVIEW

File: src/OpenLoco/src/GameCommands/Docks/CreatePort.cpp [hunk 1]
```diff
@@ -0,0 +1,502 @@
...
+            station->invalidate();
+            recalculateStationModes(_lastPlacedDockStationId);
+            recalculateStationCenter(_lastPlacedDockStationId);
+            station->updateLabel();
+            station->invalidate();
+            sub_48D794(*station);
+        }
+        if (!(flags & Flags::ghost | Flags::aiAllocated) && (flags & Flags::apply))
+        {
+            // playPlacementSound(args.pos); TODO NEED TO MOVE FUNCTION
```
Comment (duncanspumpkin) [inline src/OpenLoco/src/GameCommands/Docks/CreatePort.cpp:486]: oh lol forgot to do this

**Review by AaronVanGeffen — APPROVED** [review]
Appears to be working well. The `playPlacementSound` still needs to be sorted though.

**Comment by duncanspumpkin** [comment]
Seems to diverge one of my replays.

**Comment by duncanspumpkin** [comment]
Made a slight mistake when checking divergences last night. Turns out it does match on my replays now :)

**Pull request merged**